src/LibraryManager.Vsix/Shared/LibraryCommandService.cs

```diff
@@ -0,0 +1,352 @@
+                sw.Stop();
+
+                Logger.LogEventsSummary(results, OperationType.Clean, sw.Elapsed);
+            }
+            catch (OperationCanceledException)
+            {
+                Logger.LogEvent(LibraryManager.Resources.Text.Clean_OperationCancelled, LogLevel.Task);
+            }
+        }
+
+        private async Task RestoreInternalAsync(IDictionary<string, Manifest> manifests, CancellationToken cancellationToken)
+        {
+            Logger.LogEventsHeader(OperationType.Restore, string.Empty);
+
+            Stopwatch sw = new Stopwatch();
+            List<ILibraryOperationResult> totalResults = new List<ILibraryOperationResult>();
+
+            try
+            {
+                sw.Start();
+
+                foreach (KeyValuePair<string, Manifest> manifest in manifests)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    Project project = VsHelpers.GetDTEProjectFromConfig(manifest.Key);
+
+                    Logger.LogEvent(string.Format(LibraryManager.Resources.Text.Restore_LibrariesForProject, project?.Name), LogLevel.Operation);
+
+                    IEnumerable<ILibraryOperationResult> results = await RestoreLibrariesAsync(manifest.Value, cancellationToken).ConfigureAwait(false);
+
+                    await AddFilesToProjectAsync(manifest.Key, project, results, cancellationToken);
+                    AddErrorsToErrorList(project?.Name, manifest.Key, results);
+                    totalResults.AddRange(results);
+                }
+
+                sw.Stop();
+
+                Logger.LogEventsSummary(totalResults, OperationType.Restore, sw.Elapsed);
+                PostRestoreTelemetryData(totalResults, sw.Elapsed);
+            }
+            catch (OperationCanceledException)
+            {
+                Logger.LogEvent(LibraryManager.Resources.Text.Restore_OperationCancelled, LogLevel.Task);
+            }
+        }
+
+        private async Task<IEnumerable<ILibraryOperationResult>> RestoreLibrariesAsync(Manifest manifest, CancellationToken cancellationToken)
+        {
+            return await manifest.RestoreAsync(cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task UninstallLibraryAsync(string configFilePath, string libraryId, CancellationToken cancellationToken)
+        {
+            Logger.LogEventsHeader(OperationType.Uninstall, libraryId);
+
+            try
+            {
+                Stopwatch sw = new Stopwatch();
+                var dependencies = Dependencies.FromConfigFile(configFilePath);
+                Manifest manifest = await Manifest.FromFileAsync(configFilePath, dependencies, cancellationToken).ConfigureAwait(false);
+                IHostInteraction hostInteraction = dependencies.GetHostInteractions();
+
+                sw.Start();
```

Comment:
Just wanted to validate that the stopwatch should be starting here, not earlier in the method.